litestar/handlers/http_handlers/_utils.py

```diff
@@ -77,13 +77,10 @@ async def create_response(data: Any) -> ASGIApp:
             media_type=media_type,
             status_code=status_code,
             type_encoders=type_encoders,
+            raw_headers=raw_headers,
         )
-        response.raw_headers = raw_headers
 
-        if after_request:
-            return await after_request(response)  # type: ignore
-
-        return response
+        return (await after_request(response) if after_request else response).to_asgi_response()  # type: ignore
```

Comment:
Is this necessary?

`Response` does `await self.to_asgi_response` in its `__call__` and this expression can't resolve to anything other than `Response` or `ASGIResponse`, right?

Reply:
Oh, I meant to take the `__call__` off `Response` - I was just trying to change things gradually so that I didn't break all the tests in one go.

Reply:
Thanks, good catch!